### DISCUSSION TITLE
Deploy Flask-socketIO on IIS version 10

### DISCUSSION THREAD

**Discussion by kavindevarajan** [miguelgrinberg/Flask-SocketIO#1778] [Q&A]
Hi, I have created a python project to update a real time status for client side angular application. Below attached python code is working fine when doing under CMD-admin. But when I tried to deploy it on IIS I'm facing CORS issue. Even this CORS policy issue also raised while testing in CMD, at that time I added cors_aollowed_origins in `socket = SocketIO(app, cors_allowed_origins='*')` then it worked fine in CMD. But while doing with IIS no it's not working!!
I also installed CORS module in IIS using web installer available in IIS. And I referred how to enable CORS module in `web.config` file located at my windows server machine. Link to refer - [IIS CORS module Configuration Reference](https://docs.microsoft.com/en-us/iis/extensions/cors-module/cors-module-configuration-reference#:~:text=The%20Microsoft%20IIS%20CORS%20Module,applications%20support%20the%20CORS%20protocol.).
Due to an enterprise requirements I must have to deploy it on Windows IIS not on linux machine!
To configure python Flask application in my machine I referred this link - [Deploy python FLask application in IIS](https://medium.com/@dpralay07/deploy-a-python-flask-application-in-iis-server-and-run-on-machine-ip-address-ddb81df8edf3). I followed this link to setup my another two flask applications in IIS and both are working fine without any issues. But while doing this same configuration setup for flask-socketIO I'm facing this CORS issue. The highlight is those two flask applications till now didn't raised this CORS policy error!!
My system details:
1) Windows Server 2019
2) IIS version 10
3) Python 3.7.9

Web-Socket code:
```python
import os
import time
# import json
import subprocess
from flask import Flask, jsonify
# from flask_cors import cross_origin, CORS
from flask_socketio import SocketIO

app = Flask(__name__)
# CORS(app, resources={"r/api/*": {"origins": "*"}})
# app.config['CORS HEADERS'] = "Content-Type"
socket = SocketIO(app, cors_allowed_origins='*')
# socket1 = CORS(app)

app.config['UPLOAD_FOLDER'] = "C:\\inetpub\\wwwroot\\iAssist_IT_support\\New IT support datasets"
success = False


@app.route('/status')
def home():
    return jsonify("Hello, This is an API to update a real time response")


@socket.on('connect')
def connected():
    print('connect')


@socket.on('message', namespace='/status')
def handle_message(data):
    global success
    print("Received filename: ", data["data"])
    filenames = data["data"]
    print(filenames)
    time.sleep(3)
    success = not success
    print(success)
    socket.emit('receiver', {'Success1': 'New data merged with existing datasets'}, namespace='/status')
    time.sleep(3)
    if success:
        os.chdir("C:\\inetpub\\wwwroot\\websocket_response")
        print(os.getcwd())
        socket.emit('receiver', {'Success2': "New model training is in progress don't upload new file"},
                    namespace='/status')
        print("message2 passed")
        time.sleep(3)
        # command = "python main.py"
        # os.system(command)
        subprocess.run("python C:\\Users\\kavin\\Documents\\IT_support_chatbot-master\\Python_files\\main.py",
                       shell=True)
        print("model created")
        print(os.getcwd())
        time.sleep(3)
        socket.emit('receiver', {'Success3': "Model created successfully for sent file %s" % filenames},
                    namespace='/status')
    else:
        socket.emit('receiver', "No any progress")


if __name__ == "__main__":
    socket.run(app)
```
Web.config file:
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <handlers>
            <add name="Websocket_response" path="*" verb="*" modules="FastCgiModule" scriptProcessor="C:\Python\python.exe|C:\inetpub\wwwroot\websocket_response\wfastcgi.py" resourceType="Unspecified" requireAccess="Script" />
        </handlers>
    </system.webServer>
</configuration>
```
Please guide me how to overcome this CORS issue as soon as possible!! Probably your solution will also helpful to so many developers!!

Thank you.

**Comment by miguelgrinberg** [comment]
IIS is not a supported platform for Flask-SocketIO. I wasn't aware of any CORS issues, but many people have tried to use this web server and found no way to make WebSocket work. Unfortunately I do not use IIS so I cannot help troubleshoot.
